components/shapediver/appbuilder/actions/AppBuilderActionCameraComponent.tsx

```diff
@@ -0,0 +1,158 @@
+import AppBuilderActionComponent from "@AppBuilderShared/components/shapediver/appbuilder/actions/AppBuilderActionComponent";
+import {useViewportId} from "@AppBuilderShared/hooks/shapediver/viewer/useViewportId";
+import {useShapeDiverStoreViewport} from "@AppBuilderShared/store/useShapeDiverStoreViewport";
+import {
+	IAppBuilderActionPropsCamera,
+	isAnimateCameraAction,
+	isResetCameraAction,
+	isSetCameraAction,
+} from "@AppBuilderShared/types/shapediver/appbuilder";
+import {CAMERA_TYPE} from "@shapediver/viewer.viewport";
+import {vec3} from "gl-matrix";
+import React, {useCallback, useState} from "react";
+
+type Props = IAppBuilderActionPropsCamera & {
+	namespace: string;
+};
+
+/**
+ * Functional component for a "camera" action.
+ *
+ * @returns
+ */
+export default function AppBuilderActionCameraComponent(props: Props) {
+	const {label = "Start camera", icon = "tabler:video", tooltip} = props;
```

Comment:
@MajorMeerkatThe3rd we could set the label depending on the type of camera action, i.e. "Reset camera", "Set camera", "Animate camera"